### PR TITLE
Drop NotificationActionDescriptionComponent

### PR DESCRIPTION
**Pull request by saraycp**
This is one PR in a series of PRs towards the Notifications refactoring to STI.

The content is moved to the NotificationHelper and used from NotificationComponent until we have the final STI sub-classes.

BsRequestActionSourceAndTargetComponent is not removed because it's still used in the watchlist.

### CODE REVIEW

File: src/api/app/helpers/webui/notification_description_helper.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
+module Webui::NotificationDescriptionHelper
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # TODO: Content of ViewComponent. Move to sub-classes once STI is set.
+  def description(notification)
```
Comment (danidoni) [inline src/api/app/helpers/webui/notification_description_helper.rb:4]: Why not move this description into each object, in the same fashion @srbaker did [here](https://github.com/openSUSE/open-build-service/pull/16326/commits/ed112b35c2ab3ba36acc78441ecf49efb476dd20#diff-2c70c5303a618d03dd5d367484732e3ec2305b0354eb890c62bd28085e6251d8R6)

Comment (saraycp) [reply src/api/app/helpers/webui/notification_description_helper.rb:4]: This is only the first step towards that, which is already implemented in the draft PR.

See an example [here](https://github.com/openSUSE/open-build-service/pull/16222/files#diff-1ad94b8ff6161b4b82ffec2acd6d48aefb0e59da2dc628557956236286b2af37R1).

Comment (danidoni) [reply src/api/app/helpers/webui/notification_description_helper.rb]: Awesome! I'll shut up then :sweat_smile:

File: src/api/app/helpers/webui/notification_helper.rb [hunk 2]
```diff
@@ -1,11 +1,68 @@
+# rubocop:disable Metrics/ModuleLength
 module Webui::NotificationHelper
+  TRUNCATION_LENGTH = 100
+  TRUNCATION_ELLIPSIS_LENGTH = 3 # `...` is the default ellipsis for String#truncate
```
Comment (danidoni) [inline src/api/app/helpers/webui/notification_helper.rb:4]: We're [duplicating code](https://github.com/openSUSE/open-build-service/blob/9081ff565bd2afa5af48bf179e23f12f12f9f665/src/api/app/helpers/webui/notification_excerpt_helper.rb#L2)...

File: src/api/app/helpers/webui/notification_helper.rb [hunk 3]
```diff
@@ -1,11 +1,68 @@
...
+
   def link_to_show_less_or_more
     parameters = params.slice(:show_more, :state, :project).permit!
     less_or_more = parameters[:show_more] ? 'less' : 'more'
     parameters[:show_more] = parameters[:show_more] ? nil : '1'
     link_to("Show #{less_or_more}", my_notifications_path(parameters))
   end
 
+  # TODO: Content of ViewComponent. Move to sub-classes once STI is set.
+  def excerpt(notification)
```
Comment (danidoni) [inline src/api/app/helpers/webui/notification_helper.rb:14]: Shouldn't we use the newly introduced `NotificationExcerptHelper` for this?

Comment (danidoni) [reply src/api/app/helpers/webui/notification_helper.rb:14]: Oh I see you are removing it...

**Review by hennevogel — APPROVED** [review]

File: src/api/app/helpers/webui/notification_helper.rb (hunk 3 above)
Comment (saraycp) [reply src/api/app/helpers/webui/notification_helper.rb:14]: Yes! I introduced another commit removing the NotificationExcerptHelper. I'm finally not going to add a helper per method, but everything goes to NotificationHelper.

File: src/api/app/helpers/webui/notification_helper.rb (hunk 2 above)
Comment (saraycp) [reply src/api/app/helpers/webui/notification_helper.rb:4]: I introduced another commit removing the NotificationExcerptHelper. This code is moved from there to NotificationHelper.

**Comment by saraycp** [comment]
I rebased and applied the changes in https://github.com/openSUSE/open-build-service/pull/16337 in the NotificationHelper.

**Comment by saraycp** [comment]
I've added helper specs (based on the removed components' specs).

Any tips to make CodeClimate happy?

**Comment by saraycp** [comment]
Merge ignoring codeclimate/total-coverage, we can't figure out what's going on.

**Pull request merged**